test_conformance/spirv_new/main.cpp

```diff
@@ -174,9 +175,40 @@ test_status checkAddressWidth(cl_device_id id)
   return TEST_PASS;
 }
 
+void printUsage() {
+    log_info("Test spirv_new read binary spv files from default path spirv_bin\n");
+    log_info("In case you want to set other directory user paramter --testCachePath\n");
+}
+
 int main(int argc, const char *argv[])
 {
     gReSeed = 1;
+    bool userModifyCachePath = false;
+    for (int i = 0; i < argc; ++i) {
+        int argsRemoveNum = 0;
+        if (strcmp(argv[i], "--testCachePath") == 0) {
+            if (argv[i + 1] == NULL) {
+                log_error("Missing value for --testCachePath argument\n");
+                return TEST_FAIL;
+            } else {
+                testCachePath = std::string(argv[i + 1]);
+                argsRemoveNum += 2;
+                userModifyCachePath = true;
+            }
+        }
+
+        if (argsRemoveNum > 0) {
+            for (int j = i; j < (argc - argsRemoveNum); ++j)
+                argv[j] = argv[j + argsRemoveNum];
+
+            argc -= argsRemoveNum;
+            --i;
```

Comment:
In my opinion this is correct. See that if i == 1 ten we could got -1. The meaning of i in this place is next parameter to process. Because 2 arguments were removed we have to process again the same element in array in the next loop.

Reply:
Yes, you're right. That is correct.